[IMP] website/ecommerce: update catalog page (#15569)
odoo/documentation: content/applications/websites/ecommerce/products/catalog.rst

===============
Product catalog
===============

The eCommerce catalog displays products for customers to browse. It is organized using product
categories, available product variants, sorting options, and navigation paths. Essentially, the
eCommerce catalog is the shop page of your website.

The product catalog includes a :ref:`top bar <ecommerce/catalog/top-bar>`, a :ref:`side panel
<ecommerce/catalog/side-panel>`, and a :ref:`product listing area
<ecommerce/catalog/product-listing>`. Additionally, the :ref:`layout
<ecommerce/catalog/customize-layout>` can be customized per shop and category, and you can add
:ref:`building blocks <ecommerce/catalog/building-blocks>`.

To customize it, go to the shop page, click :guilabel:`Edit` in the upper-right corner, select the
section you want to change, and navigate to the :guilabel:`Style` tab.

.. _ecommerce/catalog/top-bar:

Top bar
=======

It possible to display a search bar or sort-by options in the top bar of the shop page by toggling
the :guilabel:`Sort By` and/or :guilabel:`Search` buttons next to the :guilabel:`Toolbar` label.

> Enable the :guilabel:`Floating` switch to make those options follow you when scrolling, and select

Reviewer: ```suggestion
Toggle the :guilabel:`Floating` switch on to make the search bar, pricelist selector, and sort-by options always visible, even when scrolling, and select
```
I'm not super convinced with my suggestion and tbh I don't really know how to best describe this "floating" option, but:
- It doesn't really "follow you", it goes to the bottom of the screen until you reach the bottom of the page (which *I think* we usually refer to as "sticky")
- I noticed the pricelist selector also "floats" so I'd include it
- toggle the switch on/off is what we usually use but I might be nitpicking here

a :guilabel:`Default Sort` option:

- :guilabel:`Featured`
- :guilabel:`Newest Arrivals`
- :guilabel:`Name (A-Z)`
- :guilabel:`Price - Low to High`
- :guilabel:`Price - High to Low`

To create a :ref:`category <ecommerce/categories_variants/categories>` quick access in the top bar,
click the :guilabel:`Top` button next to :guilabel:`Categories`, and choose a :guilabel:`Style`

> for them.

Reviewer: I would actually add a screenshot here to clarify what we're talking about (I know, shocking! 😆)

Reviewer: an example of category quick access (the "result" on a website page; website editor screenshots are rarely useful IMO)

Author: Sure thanks!

.. tip::
   - The default sort applies to *all* :ref:`categories <ecommerce/categories_variants/categories>`.
   - If you do not want to display a top bar or :ref:`side panel <ecommerce/catalog/side-panel>`,

>      disable all related options in the website editor, and make sure to only assign one pricelist

Reviewer: Should we add a link for pricelist?

     to the website.

.. _ecommerce/catalog/side-panel:

Side panel
==========

> The side panel provides advanced filtering tools to organize the catalog.

Reviewer: While you're not wrong (and this is kind of what the previous version of this doc said), I think since this section now focuses on the frontend, I would rephrase this to make it a bit more customer-focused, if that makes sense?
```suggestion
The side panel provides advanced filtering tools to help customers search and navigate the product catalog.
```

Enable :guilabel:`Filters` in the :guilabel:`Sidebar`, or set the :guilabel:`Filters` option to
:guilabel:`Off-screen menu` to add a :icon:`fa-sliders` :guilabel:`Filters` button that displays a

> filters bar when clicked. You can also :guilabel:`Hide` it entirely.

Reviewer: ```suggestion
Set the :guilabel:`Filters` field to:
- :guilabel:`Sidebar`: to display filters in the side panel;
- :guilabel:`Off-screen menu`: to display filters in a slide-out menu that opens when clicking the :icon:`fa-sliders` :guilabel:`Filters` button;
- :guilabel:`Hide`: to hide filters entirely.
```

> By default, there are two :guilabel:`Filters` enabled:

Reviewer: ```suggestion
By default, two filters are enabled:
```
I try to avoid "there are" when possible and since we're talking about the "general concept" of filters, not just the UI element, I would not use guilabel (but I will admit the distinction between a general concept and a UI element is somewhat subjective and open to interpretation).

  - :guilabel:`Price`: Use the switch to display a :guilabel:`Price Range` bar, which
    allows customers to filter products according to a specific price range by dragging adjustable
    handles.
  - :guilabel:`Tags`: Toggle the switch to display product :guilabel:`Tags` on the shop page, and
    allow customers to filter products using those tags by going to the
    :guilabel:`Tags` section in the side panel.

    .. tip::
       To create product tags on your e-commerce, go to :menuselection:`eCommerce --> Product Tags`,
       and click :guilabel:`New`. Add a :guilabel:`Tag` name, toggle the :guilabel:`Visible to
       customers` switch, and choose a :guilabel:`Color`. Optionally, add an image.

You can also add a :guilabel:`Rental Date` option to display a date range calendar to check
the availability of rental products over a specific period. The :doc:`Rental app

> <../../../sales/rental>` must be installed to use this feature.

Reviewer: ```suggestion
For :doc:`rental products <../../../sales/rental>`, you can also add a :guilabel:`Rental Date` option to display a date range calendar to check their availability over a specific period. 
```

To display :ref:`categories <ecommerce/categories_variants/categories>` in the side panel, enable
the :guilabel:`Sidebar` option next to :guilabel:`Categories`. When selected, the option
:guilabel:`Collapse Category Recursive` appears, allowing you to collapse the category in the side
panel.

.. note::
   The category filters located in the sidebar work autonomously, and are not affected by the
   :guilabel:`Filters` setting.

Attributes
----------

When you offer :ref:`product variants <ecommerce/categories_variants/product-variants>` with
different :ref:`attributes <ecommerce/categories_variants/attributes>`, those options appear
in the side panel when the :guilabel:`Filters` setting is enabled. The side panel automatically
adapts to the number of attribute sections:

- **Up to 4 attributes**: all the attribute values are displayed.
- **More than 4 attributes**: the sections are automatically collapsed.

For :guilabel:`Radio` and :guilabel:`Select` :ref:`display types <products/variants/attributes>`,
a :guilabel:`View More` button appears when there are more than **8** values, and a search field

> when the number of values exceeds **20**.

Author: Oh yes, sorry! I read it all trough and didn't realise, sorry!

.. note::
   - At least two attribute values are required for the filter to be visible.
   - The attribute filters located in the sidebar do not work autonomously, and are affected by the
     :guilabel:`Filters` setting.

.. seealso::
   - :doc:`../../../sales/sales/products_prices/products/variants`
   - :doc:`categories_variants`

.. _ecommerce/catalog/product-listing:

Product listing area
====================

.. _ecommerce/catalog/layout:

> Define the number of products to display on your shop pages by configuring the :guilabel:`Size`

Reviewer: Techncially, I think there's only one shop page? Or do you mean on the shop page and the product category pages as well?

:ref:`Add to cart <ecommerce/checkout/add-to-cart>`, :icon:`fa-exchange` :ref:`Compare
<ecommerce/products/additional_features/product-comparison>`, or :icon:`fa-heart-o` :ref:`Add to
wishlist <ecommerce/products/additional_features/wishlists>`. Define how they should be display,
i.e.: :guilabel:`Inline`, :guilabel:`On Hover`, :guilabel:`Fixed` or :guilabel:`Fixed - On Hover`.
Lastly, define a style for the buttons: :guilabel:`Subtle`, :guilabel:`Promote`
:icon:`fa-shopping-cart`, or :guilabel:`Theme colors`.

Variants and attributes
***********************

Configure :ref:`attributes <ecommerce/categories_variants/attributes>` to make them
:guilabel:`Visible` on the product card on the shop page.

You can also add a variant picture on the product card. To do so, go to the product form, click
the :icon:`fa-sitemap` :guilabel:`Variants` :ref:`smart button
<products/variants/variants-smart-button>`, select a variant, and click the
:icon:`fa-pencil` :guilabel:`(Edit)` icon to add a picture for each variant. Make sure that the
:ref:`Show Thumbnails <ecommerce/categories_variants/attributes>` is enabled on the
attribute form.

.. image:: catalog/catalog-show-thumbnails.png

>    :alt: Variant thumbnails on product card.

Reviewer: I would also remove this section since we're mostly talking about product configuration here) and move the relevant info (e.g. show thumbnails) to the Variants and categories doc page?

Author: Sorry, I might not have been clear :) I added the information about the product page because of your "in the filters?" question :) You cannot display the variant image in the filters, only attributes with the image display type. But the actual variant image can be shown on the product card in the shop page and on the product page.

In a nutshell: I will move the content and add a tip here.

.. _ecommerce/catalog/images:

Images
~~~~~~

Click the :icon:`fa-paint-brush` :guilabel:`(paint brush)` icon next to :guilabel:`Products Design`,
navigate to the :guilabel:`Images` section, and customize the image layout:

- :guilabel:`Image Ratio`: Define the ratio of images on the shop page.
- :guilabel:`Auto-crop`: Cut or trim images automatically to make them fit the page.
- :guilabel:`Hover Effect`: Define the zoom when hovering over the image.
- :guilabel:`Secondary Image`: Show a second image when hovering over the product image.

.. tip::
   :guilabel:`Replace` a media directly from the shop page using the :guilabel:`Image`
   section of the :guilabel:`Style` tab.

.. _ecommerce/catalog/customize-layout:

> Layout customization

Reviewer: For this particular section, i would change the structure and heading as follow:

(h2) Shop and category page design
Then have a small intro sentence saying: The design of the shop and individual category pages can be customized as follows:

- Choose whether the :guilabel:`Content Width` of the shop and category pages should be :guilabel:`Regular` or :guilabel:`Full width`.
- Define the headers: navigate to the relevant page, then change the header directly on the page if needed. Toggle the :guilabel:`Show Title` and/or the :guilabel:`Center Content` switches. For product category pages, toggle the :guilabel:`Show Description` switch to show the :ref:`category description <ecommerce/categories_variants/categories>`. 
- Use :doc:`building blocks <../../website/web_design/building_blocks>` to add content in the top/bottom section of the page --> add the rest of the content from lines 236-252, including the screenshot!)

--> for the headers:  this avoids repeating the same info for the shop page + category pages and only keeps what is necessary imo

====================

Customize the layout of the entire shop page by choosing the :guilabel:`Regular` or
:guilabel:`Full` :guilabel:`Content Width`.

.. tip::
   It is also possible to customize individual :ref:`product pages
   <ecommerce/products/product-form>`.

You can also define different headers for the shop and each category.

.. tabs::

   .. tab:: Shop

      Define the :guilabel:`Shop Header` by changing the title, and toggling the
      :guilabel:`Show Title` and/or the :guilabel:`Center Content` switches.

   .. tab:: Category

>       Define the :guilabel:`(Category) Header` by changing the title, and toggling the

Reviewer: Nitpick and quite obvious (but maybe not for everyone, hence this comment!) but I'd add that you first need to navigate to the relevate product category page

===============
Product catalog
===============

The eCommerce catalog displays products for customers to browse. It is organized using :ref:`product
categories <ecommerce/categories_variants/categories>`, available :ref:`product variants
<ecommerce/categories_variants/product-variants>`, sorting options, and navigation paths.
Essentially, the eCommerce catalog is the shop page of your website.

The product catalog features a :ref:`toolbar and some filters <ecommerce/catalog/toolbar-and-filters>`
in the sidebar and/or top bar, along with a dedicated :ref:`product listing area
<ecommerce/catalog/product-listing>`. Additionally, the shop and category pages' :ref:`layout
<ecommerce/catalog/customize-layout>` as well as the :ref:`eCommerce footer

> <ecommerce/catalog/footer>` can be customized, and you can add :ref:`building blocks

Reviewer: ```suggestion
<ecommerce/catalog/customize-layout>`, as well as the :ref:`eCommerce footer
<ecommerce/catalog/footer>`, can be customized, and you can add :ref:`building blocks
```

search and navigate the product catalog using :ref:`categories <ecommerce/catalog/categories>` and
:ref:`attributes <ecommerce/catalog/attributes>`.

.. _ecommerce/catalog/categories:

Categories
----------

To create a :ref:`category <ecommerce/categories_variants/categories>` quick access in the toolbar
or general access in the sidebar, click the :guilabel:`Top` and/or :guilabel:`Sidebar` button next
to :guilabel:`Categories`, and choose a :guilabel:`Style` for them. When the :guilabel:`Sidebar` is
selected, the option :guilabel:`Collapse Category Recursive` appears, allowing you to collapse the
category in the sidebar.

.. image:: catalog/category-quick-access-top.png
   :alt: Category quick access in the toolbar.

.. note::
   The category filters located in the sidebar are not affected by the :guilabel:`Filters` setting.

.. _ecommerce/catalog/toolbar:

Toolbar
-------

It is possible to display a search bar or sort-by options in the toolbar of the shop page by
clicking the :guilabel:`Sort By` and/or :guilabel:`Search` buttons next to the :guilabel:`Toolbar`
label. Toogle the :guilabel:`Floating` switch on to always display the search bar, the

> :ref:`pricelist selector <ecommerce/prices/pricelists>` (if any), and sort-by options at the botton

Reviewer: ```suggestion
:ref:`pricelist selector <ecommerce/prices/pricelists>` (if any), and sort-by options at the bottom
```

.. note::
   The number of products displayed on a page may affect performance and page loading speed.

In addition, you can manually change a product’s position on the shop page or product category page.
To do so, click a specific product, change its :guilabel:`Size`, and :guilabel:`Re-order` it using
the arrows in the :guilabel:`Product` section of the :guilabel:`Style` tab. The
:icon:`fa-angle-double-left` (:guilabel:`double left arrow`) and :icon:`fa-angle-double-right`
(:guilabel:`double right arrow`) buttons allow you to move the product to the extreme left or right,
and the :icon:`fa-angle-left` (:guilabel:`single left arrow`) and :icon:`fa-angle-right`
(:guilabel:`single left arrow`) buttons allow you to move it one row to the left or right. You can
also choose or create a :ref:`ribbon or badge
<ecommerce/products/additional_features/product-highlight>` to highlight a specific product.

.. tip::
   It is also possible to change the products' positions on the shop page by going to
   :menuselection:`Website --> eCommerce --> Products`, switching to the list view, and
   dragging and dropping the products within the list.

.. _ecommerce/catalog/products-design:

> Products design

Author: I thought to use "Products design" because all these options are available in the customization menu of the "Products Design" (when clicked on the paintbrush). 
But we can also change it.

---------------

Choose a :guilabel:`Products Design`, click the :icon:`fa-paint-brush` :guilabel:`(paint brush)`

> icon next to it to customize the product cards' design, then:

Reviewer: I think I'd change this slightly to start with the purpose so we have a small intro:

```suggestion
To customize the product cards' design, choose a :guilabel:`Products Design`, then click the :icon:`fa-paint-brush` :guilabel:`(paint brush)` icon next to it to:
```

Depending on the selected template, you can :ref:`customize <website/building_blocks/edit>` the
blocks in the :guilabel:`Style` tab.

.. example::
   .. image:: catalog/catalog-products-block.png
      :alt: Example of a catalog block for specific products

   For example, for this specific :guilabel:`Catalog` block template to display a selection of
   products sold on your website, you can:

   - Adapt the block's design.
   - Select a :guilabel:`Cards Design`, and click the :icon:`fa-paint-brush` :guilabel:`(paint
     brush)` icon to :ref:`customize <ecommerce/catalog/products-design>` it.
   - Define which items are shown in the block by selecting the relevant :guilabel:`Filter` option,
     e.g., :guilabel:`Newest Products` to show the latest arrivals.
   - Select a specific product category in the :guilabel:`Category` dropdown list.
   - Filter products by toggling the :guilabel:`Tags` and/or :guilabel:`Show variants` options.

.. _ecommerce/catalog/footer:

Ecommerce-specific website footer
=================================

The :ref:`website footer <website/header_footer/footer-design>` can also be configured to use

> the ecommerce-specific template. Then add the relevant :ref:`links <website/elements/links>`, and

Reviewer: ```suggestion
the ecommerce-specific template. Then, add the relevant :ref:`links <website/elements/links>`, and
```

   - eCommerce categories can also be added as :ref:`mega menu items
     <website/header_footer/mega-menus>` for quick access.

Catalog blocks
~~~~~~~~~~~~~~

The :guilabel:`Catalog` :doc:`building blocks
</applications/websites/website/web_design/building_blocks>` are used to create quick access to
your e-commerce shop. You can showcase your categories, a specific product selection, or a banner
with general information or promotions.

Depending on the selected layout, you can customize the blocks in the :guilabel:`Style` tab.

.. example::
   Add a building block to display a selection of products sold on your website.

   .. image:: catalog/catalog-product-block.png

>       :alt: Example of a catalog block for specific products

Author: ok

   By default, the block displays the :guilabel:`Newest Products`. To change which products are
   shown, go to the :guilabel:`Style` tab's :guilabel:`Products` section and set the
   :guilabel:`Filter` field to :guilabel:`Recently Sold Products` or :guilabel:`Recently Viewed
   Products (per user)`. In addition, it is possible to display products from a specific category
   only using the :guilabel:`Category` field.

   You can also filter products by :guilabel:`Tags`, :guilabel:`Show variants`, and adjust the

>    display by selecting a different :guilabel:`Cards Design`.

Author: Ok thanks!
